src/ray/core_worker/core_worker.cc

```diff
@@ -222,7 +221,7 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
   // driver creates an object that is later evicted, we should notify the
   // user that we're unable to reconstruct the object, since we cannot
   // rerun the driver.
-  if (worker_type_ == WorkerType::DRIVER) {
+  if (worker_type_ == WorkerType::DRIVER || local_mode_enabled_) {
```

Comment:
when `local_mode_enabled_` is set, the type is already set to `DRIVER`, so there's no need to add the extra check here?